extension-manifest-v3/src/background/stats.js

```diff
@@ -265,15 +265,20 @@ if (__PLATFORM__ === 'safari') {
         // with the correct tabId (sometimes it is correct, sometimes it is 0).
         // Thus, let the content_script fire it.
         case 'onCommitted': {
-          const { domain, hostname } = parse(sender.url);
-          setupTabStats(sender.tab.id, domain || hostname);
+          const request = Request.fromRequestDetails({ url: sender.url });
+          setupTabStats(
```

Comment:
now I start to wonder if this is not too late for non-safari browsers. couldn't it be the some webRequests for this navigation would fire before `onCommited`, taking Service Workers into consideration

Reply:
The implementation has changed, please make another round of review and make the comment again if it still applies.